Client/Coordinators/Coordinator.swift

```diff
@@ -7,7 +7,7 @@ import Foundation
 protocol Coordinator {
     var id: UUID { get }
     var childCoordinators: [Coordinator] { get }
-    var router: Router { get }
+    var router: Router? { get }
 
```

Comment:
Isn't Router something required for a Coordinator? I'm curios why it's optional

Reply:
I wrote in the description, I made the Router optional on protocol with a conveniance init since in the case of SceneCoordinator we don't have a router yet at first. On the other coordinator we'll get the router at init